momepy/strokegraph.py

```diff
@@ -0,0 +1,370 @@
+    )
+    if angle > 90:
+        angle = 180 - angle
+    return angle
+
+
+def _get_end_segment(linestring, point):
+    """
+    Helper function for ``make_stroke_graph()``.
+    Returns the first or last two-Point segment of a LineString.
+
+    Parameters
+    ----------
+    linestring: shapely.LineString
+    point: list
+        A list of length 2 containing the coordinates of either
+        the first or the last point on the linestring.
+    """
+    point = tuple(point)
+    coords = list(linestring.coords)
+    assert point in coords, "point not on linestring!"
+    if point == coords[0]:
+        geom = [np.array(val) for val in linestring.coords[:2]]
+    elif point == coords[-1]:
+        geom = [np.array(val) for val in linestring.coords[-2:]]
+    else:
+        raise ValueError("point is not an endpoint of linestring!")
+    return np.array(geom[0] - geom[1])
+
+
+def coins_to_nx(coins):
+    """
+    Creates the stroke graph of a street network. The stroke graph is similar to, but
+    not identical with, the dual graph. In the stroke graph, each stroke (see
+    ``momepy.COINS``) is a node; and each intersection between two strokes is an edge.
+
+    Parameters
+    ----------
+    coins: momepy.COINS
+        Strokes computed from a street network.
+
+    Returns
+    -------
+    stroke_graph : Graph
+        A networkx.Graph object.
+
+    Examples
+    --------
+    >>> import geopandas as gpd
+    >>> gdf = gpd.read_file(momepy.datasets.get_path("bubenec"), layer="streets")
+    >>> gdf = momepy.remove_false_nodes(gdf)
+    >>> primal_graph = momepy.gdf_to_nx(gdf, preserve_index=True, approach="primal")
+    >>> lines = momepy.nx_to_gdf(primal_graph, points=False, lines=True)
+    >>> coins = momepy.COINS(lines)
+    >>> stroke_graph = momepy.coins_to_nx(coins)
+    """
+
+    # get strokes attributes from coins
+    stroke_attribute = coins.stroke_attribute()
+
+    # get strokes gdf from coins
+    stroke_gdf = coins.stroke_gdf()
+
+    # add representative point to stroke_gdf (for later visualization)
+    stroke_gdf["rep_point"] = stroke_gdf.geometry.apply(
+        lambda x: x.interpolate(0.5, normalized=True)
+    )
+
+    # add stroke_id column
+    stroke_gdf["stroke_id"] = stroke_gdf.index
+
+    # add column containing indeces of edges comprising each stroke
+    # (using COINS.stroke_attribute to map into ID defined in lines gdf)
+    stroke_gdf["edge_indeces"] = stroke_gdf.stroke_id.apply(
+        lambda x: list(stroke_attribute[stroke_attribute == x].index)
+    )
+
+    # recreate primal graph from coins.edge_gdf
+    graph = gdf_to_nx(coins.edge_gdf, preserve_index=True, approach="primal")
+
+    # Add stroke ID to each edge on (primal) graph
+    nx.set_edge_attributes(
+        G=graph,
+        values={
+            e: int(stroke_attribute[graph.edges[e]["index_position"]])
+            for e in graph.edges
+        },
+        name="stroke_id",
+    )
```

Comment:
I don't get the point of this. `graph` already has stroke_id on edges coming from `gdf_to_nx`. It is a wrong one or what?